src/main/java/jenkins/plugins/jobcacher/arbitrary/TarArbitraryFileCacheStrategy.java

```diff
@@ -0,0 +1,105 @@
+package jenkins.plugins.jobcacher.arbitrary;
+
+import hudson.FilePath;
+import hudson.remoting.VirtualChannel;
+import hudson.util.DirScanner;
+import hudson.util.io.ArchiverFactory;
+import jenkins.MasterToSlaveFileCallable;
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
+
+import java.io.*;
+import java.nio.file.Files;
+
+public class TarArbitraryFileCacheStrategy extends AbstractCompressingArbitraryFileCacheStrategy {
+
+    private final String compressorName;
+    private final String archiveExtension;
+
+    public TarArbitraryFileCacheStrategy(String compressorName, String archiveExtension) {
+        this.compressorName = compressorName;
+        this.archiveExtension = archiveExtension;
+    }
+
+    @Override
+    protected String getArchiveExtension() {
+        return archiveExtension;
+    }
+
+    @Override
+    protected void uncompress(FilePath source, FilePath target) throws IOException, InterruptedException {
+        source.act(new ExtractTarGzCallable(target, compressorName));
+    }
+
+    @Override
+    protected void compress(FilePath source, String includes, String excludes, boolean useDefaultExcludes, FilePath target) throws IOException, InterruptedException {
+        target.act(new CreateTarGzCallable(source, includes, excludes, useDefaultExcludes, compressorName));
+    }
+
+    private static class ExtractTarGzCallable extends MasterToSlaveFileCallable<Void> {
```

Comment:
As this contains also the ZSTD compressor, the class name should not contain `TarGz`.